ypathgen/pathgen.go

```diff
@@ -631,47 +620,38 @@ func getNodeDataMap(directories map[string]*ygen.Directory, leafTypeMap map[stri
 				HasDefault:            false,
 				YANGTypeName:          "",
 				YANGPath:              "/",
-				GoPathPackageName:     goPackageName(dir.Entry, splitByModule, trimOCPackage, packageName, packageSuffix),
+				GoPathPackageName:     goPackageName(dir.RootModule, splitByModule, trimOCPackage, true, packageName, packageSuffix),
```

Comment:
I think I see why `RootModule` was defined here - since you want to do something based on the top-level parent of the node in the schema tree. I suggest that this is better done by looking at the path to avoid the confusing definition.

You can just do `strings.Split(path, "/")[1]` to find this - and this is significantly less confusing than having it contained in the IR IMHO, since today we're implying that this thing is meaningful in some manner for all different IR users, whereas it's really just nuance of how ypathgen has chosen to name packages (another approach could have been to split it based on the defining module, if we allow all possible splits to populate the IR, it is going to grow without bounds).

Reply:
As I just commented in https://github.com/openconfig/ygot/pull/655#discussion_r859359783 I think we'll need this information somehow, so we can remove `Path` xor `RootModule`. I'm OK either way. I just realized we could call `SchemaPath` `DataPath` to distinguish it from `Path`. RFC7950 uses the term [here](https://datatracker.ietf.org/doc/html/rfc7950#section-4.2.7).

Reply:
But `DataPath` would imply that this is in the data tree - right? Such that we really would end up with things like `/interfaces/interface[name=eth0]/config/name` as the 'datatree' path for a leaf.

I wonder whether we should just create a namespace-qualified version of the path - such that we have `/openconfig-interfaces:interfaces/openconfig-interfaces:interface/...` and then this would include all of the information that one could possibly need here?

Reply:
I'm fine with keeping the name as-is, although `SchemaPath` also doesn't seem right because it doesn't have the choice/case statements. I also see the value in creating a new `Path` type to contain both the defining module as well as the node name. For example it could be used to workaround this issue: https://docs.google.com/document/d/1EUzb_uXQWnDETFJkOm2Ad9fIJ_N4rytq-f3JwupWBu8/edit#heading=h.xdjxwhqgihsl

My proposal is to leave this be for now, figure out what we're going to do with `ParsedDirectory`'s path (a possibly backwards-incompatible change), and then create this new `Path` type.

I'm recording these TODOs in my own notes so I'm not losing track of them.

Reply:
Sounds good to me.